lib/chai/utils/hasProperty.js

```diff
@@ -35,30 +33,19 @@ var type = require('type-detect');
  *     hasProperty(2, obj.arr);  // true
  *     hasProperty(3, obj.arr);  // false
  *
+ * @param {String|Symbol} name
  * @param {Object} object
- * @param {String|Number} name
  * @returns {Boolean} whether it exists
  * @namespace Utils
- * @name getPathInfo
+ * @name hasProperty
  * @api public
  */
 
-var literals = {
-    'number': Number
-  , 'string': String
-};
-
 module.exports = function hasProperty(name, obj) {
-  var ot = type(obj);
-
-  // Bad Object, obviously no props at all
-  if(ot === 'null' || ot === 'undefined')
+  if (obj === undefined || obj === null) {
```

Comment:
`undefined` can't be redefined since IE9. According to #124 and #117, IE8 is not supported.